src/Generator/Generators/CLI/CLIHeadersTemplate.cs

```diff
@@ -283,6 +286,34 @@ public void GenerateClass(Class @class)
             WriteLine("};");
         }
 
+        private bool CheckClassIsStructible(Class @class)
+         {
+             if(!@class.DeclaredStruct)
+                 return false;
+
+             if (@class.IsValueType)
+                 return true;
+             if (@class.IsInterface)
+                 return false;
+             if (@class.IsStatic)
+                 return false;
+             if (@class.IsAbstract)
+                 return false;
+             if (@class.HasBase)
```

Comment:
Are you sure this is necessary? C# structs can inherit from other classes, they just cannot be inherited from.